src/agl/web_app_manager_service_agl.cc

```diff
@@ -356,6 +358,11 @@ void WebAppManagerServiceAGL::LaunchStartupAppFromConfig() {
       description = xmlNodeListGetString(doc, node->xmlChildrenNode, 1);
     if (!xmlStrcmp(node->name, (const xmlChar*)"author"))
       author = xmlNodeListGetString(doc, node->xmlChildrenNode, 1);
+    if (!xmlStrcmp(node->name, (const xmlChar*)"window")) {
+      width = xmlGetProp(node, (const xmlChar*)"width");
+      height = xmlGetProp(node, (const xmlChar*)"height");
+    }
```

Comment:
Now that we're no longer use XMLs, do you think it is worth looking into converting this to JSON instead? We do this XMLs <-> JSON conversion anyway, and seems we kind of shoot ourselves in the foot with it.

Reply:
Yes, definitely that's something I want to do. Mostly because WAM already has appinfo.json for that purpose, and a big part of this stuff is just for converting from XML to application description (that essentially already implements appinfo.json).

This has been already reported as ticket https://jira.automotivelinux.org/browse/SPEC-4252 and will reduce dramatically the delta with upstream.

Reply:
> Could you elaborate on this? How much of xdg shell is being used right now in AGL? We support XDG shell in chromium side, and after the fix to make sure app ID is set, we are passing that information to the other layers.

Right, so xdg-shell is being now as the de facto protocol for transforming wl_surfaces to/into windows. Anything related to window creation and resizing is done with xdg-shell. Before the transition to a custom shell (and here I mean agl-shell) and to a compositor based on libweston, it was using ivi-shell, and weston itself as compositor. On the client side, like you said, it is being done in chromium, while the server side, implementation of xdg-shell happens in a abstraction library called libweston-desktop. That is also where to handled deprecated wl_shell and xwayland. 

Anyway, the shift to xdg-shell, from ivi-shell, meant that regular application, using xdg-shell happened seamlessly. 

Activation of applications (or switching to them), is not tied to a particular wl_surface because the compositor needs to have a mapping between the application and its driving surface (probably another reason could be that we could a bunch of them, like subsurfaces to child top-level surfaces). So we use an appid, an appid which applications have to set upon creating a xdg-shell top-level role. The chromium version that I've worked with, with wasn't setting the appid, so besides the client protocol implementation of the agl-shell protocol into chromium (ready() request, activate_app, set_panel/set_bg) I've also added a way to set the an appid. 

I suppose because I've added 'Agl' in its name meant that it had to with AGL per se, but it is actually related to xdg-shell. Chromium should have a way to expose function to clients, as it does with the title. There are two xdg-shell requests that aren't specific to window operations: set_title, and set_app_id. As I haven't really checked in the latest change of chromium, maybe there is already a way to do that?